test/main.cpp

```diff
@@ -217,6 +219,51 @@ void print_usage(char const* executable)
 		"by -l. If no test is specified, all tests are run\n", executable);
 }
 
+void change_directory(std::string const& f, error_code& ec)
+{
+	ec.clear();
+
+#ifdef TORRENT_WINDOWS
+#if TORRENT_USE_WSTRING
+#define SetCurrentDirectory_ SetCurrentDirectoryW
+	std::wstring n = convert_to_wstring(f);
+#else
+#define SetCurrentDirectory_ SetCurrentDirectoryA
+	std::string const& n = convert_to_native(f);
+#endif // TORRENT_USE_WSTRING
+
+	if (SetCurrentDirectory_(n.c_str()) == 0)
+		ec.assign(GetLastError(), system_category());
+#else
+	std::string n = convert_to_native(f);
+	int ret = ::chdir(n.c_str());
+	if (ret != 0)
+		ec.assign(errno, system_category());
+#endif
+}
+
+struct unit_directory_guard
+{
+	std::string dir;
+	~unit_directory_guard()
+	{
+		error_code ec;
+		change_directory(root_dir, ec); // windows will not allow to remove current dir, so let's change it to root
+		if (ec)
+		{
+			std::printf("Failed to change directory: %s\n", ec.message().c_str());
```

Comment:
this should probably be a fatal error. You should be able to use the ``TEST_ERROR`` macro for that, I would imagine.